File: 733-flood-fill/flood-fill.cpp
class Solution {
public:
int oldcolor;
int newcolor;
int m,n;
void dfs(vector<vector<int>> &img,int r,int c){
    if(r<0 || c<0 || r>=m || c>=n){
        return;
    }
    if(img[r][c]!=oldcolor){
        return;
    }
    img[r][c]=newcolor;
    dfs(img,r+1,c);
    dfs(img,r-1,c);

    dfs(img,r,c+1);

    dfs(img,r,c-1);

}
    vector<vector<int>> floodFill(vector<vector<int>>& img, int r, int c, int color) {
        m = img.size();
        n = img[0].size();
        oldcolor = img[r][c];
        newcolor = color;
                if(oldcolor==newcolor){
                    return img;
                   
                    
                }
                 dfs(img,r,c);

       
                 return img;

    }
};

File: 733-flood-fill/flood-fill.java
class Solution {
    public int newcolor;
    public int oldcolor;
    public int cnt;
    public void dfs(int [][]arr,int i,int j){
        if(i<0||j<0||i>=arr.length||j>=arr[0].length){
            return ;
        }
        if(arr[i][j]!=oldcolor){
            return ;
        }
        arr[i][j]=newcolor;
        dfs(arr,i+1,j);
        dfs(arr,i-1,j);
        dfs(arr,i,j+1);
        dfs(arr,i,j-1);
    }
    public int[][] floodFill(int[][] img, int sr, int sc, int color) {
        int n=img.length;
        int m=img[0].length;
        oldcolor=img[sr][sc];
        newcolor=color;
        if(oldcolor==newcolor){
            return img;
        }
        dfs(img,sr,sc);
        return img;
    }
}